nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -0,0 +1,209 @@
+            "        NonNullTypeParam<@Nullable String> t2 = null;",
+            "    }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void constructorTypeParamInstantiation() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.nullness.Nullable;",
+            "class Test {",
+            "    static class NonNullTypeParam<E> {}",
+            "    static class NullableTypeParam<E extends @Nullable Object> {}",
+            "    static void testOkNonNull(NonNullTypeParam<String> t) {",
+            "        NonNullTypeParam<String> t2 = new NonNullTypeParam<String>();",
+            "    }",
+            "    static void testBadNonNull(NonNullTypeParam<String> t) {",
+            "        // BUG: Diagnostic contains: Generic type parameter",
+            "       NonNullTypeParam<String> t2 = new NonNullTypeParam<@Nullable String>();",
+            "        // BUG: Diagnostic contains: Generic type parameter",
+            "        testBadNonNull(new NonNullTypeParam<@Nullable String>());",
+            "        testBadNonNull(new NonNullTypeParam<",
+            "              // BUG: Diagnostic contains: Generic type parameter",
+            "              @Nullable String>());",
+            "    }",
+            "    static void testOkNullable(NullableTypeParam<String> t1, NullableTypeParam<@Nullable String> t2) {",
+            "        NullableTypeParam<String> t3 = new NullableTypeParam<String>();",
+            "        NullableTypeParam<@Nullable String> t4 = new NullableTypeParam<@Nullable String>();",
+            "    }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void extendedClassTypeParamInstantiation() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.nullness.Nullable;",
+            "class Test {",
+            "    static class NonNullTypeParam<E> {}",
+            "    static class MixedTypeParam<E1, E2 extends @Nullable Object, E3 extends @Nullable Object, E4> {}",
+            "    // BUG: Diagnostic contains: Generic type parameter",
+            "    static class InvalidSubclass extends NonNullTypeParam<@Nullable String> {}",
+            "    // BUG: Diagnostic contains: Generic type parameter",
+            "     static class PartiallyInvalidSubclass extends MixedTypeParam<@Nullable String, String, String, String> {}",
+            "     static class ValidSubclass extends MixedTypeParam<String, @Nullable String, @Nullable String, String> {}",
+            "     static class PartiallyInvalidSubclass2 extends MixedTypeParam<String, String, String, ",
+            "         // BUG: Diagnostic contains: Generic type parameter",
+            "         @Nullable String> {}",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void subClassTypeParamInstantiation() {
```

Comment:
Good catch.  In ee57c5614bffeb89b21f767778c135c1234d308c I tweaked the test and renamed it to `multipleTypeParametersInstantiation` to clarify it is about having multiple type parameters rather than just one.